dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java

```diff
@@ -674,15 +677,34 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
 		
 	}
 
-	private void copyMetadata(final String uuid, final String fieldVarName, final DotTempFile temp)
-			throws DotDataException {
+	/**
+	 *
+	 * @param uuid
+	 * @param fieldVarName
+	 * @param temp
+	 * @throws DotDataException
+	 */
+	private void copyMetadata(final String uuid, final String fieldVarName,
+			final DotTempFile temp) {
 		//Basically here a new temp file is generated and we should transfer any custom generated metadata
-		if(UtilMethods.isSet(fieldVarName) && UtilMethods.isSet(uuid)){
-		   final Optional<Metadata> metadata = fileMetadataAPI.getMetadata(uuid);
-		   if(metadata.isPresent()){
-			  fileMetadataAPI.putCustomMetadataAttributes(temp.id, ImmutableMap
-					  .of(fieldVarName, metadata.get().getCustomMeta()));
-		   }
+		if (UtilMethods.isSet(fieldVarName) && UtilMethods.isSet(uuid)) {
+			try {
+				if (UUIDUtil.isUUID(uuid)) {
+					final Contentlet content = contentAPI
+							.find(uuid, APILocator.systemUser(), false);
+					final Metadata metadata = fileMetadataAPI.getMetadata(content, fieldVarName);
+					fileMetadataAPI.putCustomMetadataAttributes(temp.id, ImmutableMap

```

Comment:
not sure if it is a good question but; why temp.id is the key

Reply:
The image editor frequently uses temp files 
Like this http://localhost:8080/contentAsset/image/temp_75f354d10e/fileAsset
in that scenario, we wouldn't have a valid inode So we must work with a temp id